sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ConditionalPreconditionsTest.java

```diff
@@ -651,6 +657,535 @@ void testDescriptionModificationModificationNoDescriptors() {
+            presenceMap.put(invocation.getArgument(0), true);
+            return ResponseTypes.Result.RESULT_SUCCESS;
+        });
+        when(mockManipulations.removeDescriptor(anyString())).thenAnswer((Answer<ResponseTypes.Result>) invocation -> {
+            presenceMap.put(invocation.getArgument(0), false);
+            return ResponseTypes.Result.RESULT_SUCCESS;
+        });
+        when(mockManipulations.triggerDescriptorUpdate(anyString()))
+                .thenAnswer((Answer<ResponseTypes.Result>) invokation -> {
+                    updateMap.put(invokation.getArgument(0), true);
+                    return ResponseTypes.Result.RESULT_SUCCESS;
+                });
+        when(testClient.getSdcRemoteDevice().getMdibAccess().getDescriptor(anyString()))
+                .thenAnswer((Answer<Optional<AbstractDescriptor>>) invocation -> {
+                    final String handle = invocation.getArgument(0);
+                    if (presenceMap.get(handle)) {
+                        return Optional.of(mock(AbstractDescriptor.class));
+                    } else {
+                        return Optional.empty();
+                    }
+                });
+
+        // when
+        assertTrue(
+                ConditionalPreconditions.DescriptionModificationMdsDescriptorPrecondition.manipulation(testInjector));
+
+        // then
+        final var insertCaptor = ArgumentCaptor.forClass(String.class);
+        final var removeCaptor = ArgumentCaptor.forClass(String.class);
+        final var updateCaptor = ArgumentCaptor.forClass(String.class);
+        verify(mockManipulations, times(1)).getRemovableDescriptorsOfClass(MdsDescriptor.class);
+        verify(mockManipulations, times(1)).insertDescriptor(insertCaptor.capture());
+        verify(mockManipulations, times(1)).removeDescriptor(removeCaptor.capture());
+        verify(mockManipulations, times(1)).triggerDescriptorUpdate(updateCaptor.capture());
+
+        assertEquals(descriptor1Handle, insertCaptor.getValue());
+        assertEquals(descriptor1Handle, removeCaptor.getValue());
+        assertEquals(descriptor1Handle, updateCaptor.getValue());
+    }
+
+    /**
+     * Tests whether DescriptionModificationMdsDescriptorPrecondition correctly calls manipulation
+     * in case all RemovableDescriptors are absent.
+     */
+    @Test
+    @DisplayName("DescriptionModificationMdsDescriptorPrecondition correctly calls manipulation when all "
+            + "RemovableDescriptors are absent.")
+    public void testDescriptionModificationMdsDescriptorPreconditionGoodAbsentDescriptors() {
+
+        // given
+        final var descriptor1Handle = "superHandle1";
+        final var descriptor2Handle = "handle;Süper;2";
+
+        final var presenceMap = new HashMap<>(Map.of(
+                descriptor1Handle, false,
+                descriptor2Handle, false));
+        final var updateMap = new HashMap<>(Map.of(
```

Comment:
This HashMap is updated but otherwise not used for the test case.

Reply:
Removed.